server/src/main/scala/drt/server/feeds/lgw/LGWFeed.scala

```diff
@@ -124,24 +130,46 @@ case class LGWFeed(certPath: String, privateCertPath: String, namespace: String,
           rawICAO = (n \\ "AirlineICAO" text) + parseFlightNumber(n),
           rawIATA = (n \\ "AirlineIATA" text) + parseFlightNumber(n),
           Origin = parseOrigin(n),
-          SchDT = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text ).equals("SCT") ).map(n=>n text).getOrElse(""),
-          Scheduled = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text ).equals("SCT") ).map(n=> services.SDate.parseString(n text).millisSinceEpoch).getOrElse(0),
+          SchDT = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text).equals("SCT")).map(n => n text).getOrElse(""),
+          Scheduled = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("ONB") && (n \ "@TimeType" text).equals("SCT")).map(n => services.SDate.parseString(n text).millisSinceEpoch).getOrElse(0),
           PcpTime = 0,
           LastKnownPax = None)
         log.info(s"parsed arrival: $arrival")
-        arrival
+        (arrival, locationOption)
       }
 
       result.toList
     }
 
+    val processDeleteIfApplicable: List[(Arrival, Option[String])] => List[Arrival] = { arrivalsAndLocationList =>
+      val deletePipeline = (
+        addHeader("Authorization", wrapHeader)
+          ~> sendAndReceive
+        )
+      arrivalsAndLocationList.flatMap(_._2).foreach { location =>
+        deletePipeline(Delete(location)).recoverWith {
+          case t: Throwable =>
+            log.warn(s"Failed to send a Delete request to url $location", t)
+            Future(location)
+        }
+      }
+
+      arrivalsAndLocationList.map(_._1)
+    }
+
     val resultPipeline: pipelining.WithTransformerConcatenation[HttpRequest, Future[List[Arrival]]] = (
-      addHeader("Authorization", wrapHeader)
-        ~> ourSendReceive
+      addHeaders(List(RawHeader("Authorization", wrapHeader), RawHeader("MessageId", UUID.randomUUID().toString)))
```

Comment:
what is the purpose of this random message id?

Reply:
After now reading the documentation it doesn't have it on this POST, so I'll remove it.

The use the message Id in other HTTP request such that :
>"In order to enable duplicate-message detection, an ID must be set on the message by adding the MessageId property to the BrokerProperties collection. The value of the ID should be unique for the message such that if it’s sent a second time, the duplicate will be detected and ignored."